### ISSUE TITLE
Keyword assert does not appear in the Modelica grammar

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#897]
**Reported by rfranke on 2 Nov 2012 09:12 UTC**
ModelicaSpec33.pdf defines assert as keyword in section 2.3.3. Opposed to all other keywords, assert does not appear in the grammar though (see Appendix B.2). Section 8.3.5.2 "Restrictions on Equations withing When-Equations" says that assert would be an operator. Section 8.3.7 "assert" introduces it as special equation or statement.

Shouldn't assert be taken out of the list of keywords in section 2.3.3?

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/897

**Comment by modelica-trac-importer** [comment]
**Comment by choeger on 2 Nov 2012 10:08 UTC**
I agree. It does not even occur in the grammar. So in theory you cannot write an assert statement in Modelica 3.3 ;).

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 2 Nov 2012 10:12 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 2 Nov 2012 14:10 UTC**
I think it is correct that it doesn't appear in the grammar, and should be removed from the list of keywords. Both MapleSim, and the Dymola demo version handed out at Modelica 2009, accept the following model:

```mo
model Main
    Real assert;
equation
    assert = 2;
end Main;
```

Thus neither treat `assert` as a keyword.

In the examples in the specification itself, sometimes `assert` has keyword syntax highlighting, and sometimes it does not.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 20 Dec 2012 13:47 UTC**
Housekeeping, defaulting all new tickets to [milesstone:\*undecided\*] (is normally done automatically)

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 9 Mar 2016 14:41 UTC**
Resolved by #1163 where assert was removed from keyword-list.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 9 Mar 2016 17:21 UTC**
